thunder/core/update_aliases.py

```diff
@@ -0,0 +1,107 @@
+from collections import defaultdict
+from itertools import chain
+from functools import reduce, partial
+
+import thunder.core.prims as prims
+from thunder.core.proxies import TensorProxy, variableify, unvariableify
+from thunder.core.pytree import tree_flatten, tree_unflatten
+from thunder.core.symbol import BoundSymbol
+from thunder.core.trace import from_trace, tracectx, TraceCtx as Trace, TraceProvenance
+
+
+def _get_update_bsym(group, swap_map, new_aliases):
+    aliases = tuple(unvariableify(alias) for alias in group)
+    update_aliases_bsym = prims.update_aliases.bind(aliases, output=new_aliases)
+    update_aliases_bsym = update_aliases_bsym.from_bsym_swap_proxies(swap_map)
+    for new_alias, old_alias in zip(new_aliases, group):
+        swap_map[old_alias] = new_alias
+    return update_aliases_bsym, swap_map
+
+
+def _get_new_alias(alias, trace):
+    with tracectx(trace):
+        # !! better to use the meta function?
+        new_alias = TensorProxy(like=alias.proxy, requires_grad=alias.proxy.requires_grad)
+    return new_alias
+
+
+def _get_new_aliases(aliases, trace):
+    return tuple(map(partial(_get_new_alias, trace=trace), aliases))
+
+
+def _is_inplace_op(bsym):
+    return bsym.sym.tags and prims.OpTags.IN_PLACE in bsym.sym.tags
+
+
+def _is_view_creation_op(bsym):
+    import thunder.torch as ltorch
+
+    # !!! check on _syms_that_may_return_views?
+    return bsym.sym in ltorch._syms_returning_views
```

Comment:
I guess this might be orthogonal to this PR, but what's the approach we're to take for the ops listed in `_syms_that_may_return_views`?